### PR TITLE
Initialization additions to VM

### PR DESCRIPTION
**Pull request by rgolangh**
1. Add support for Ignition based OSs RHCOS/FCOS
   (it used to piggy-back on cloud-init)

2. Add support for Sysprep (windows)

3. Add support for VM "type" field (Server/Desktop/High performance VMs)
   More on High performance VM: https://www.ovirt.org/develop/release-management/features/virt/high-performance-vm.html

Acceptance test:
```
    make testacc TESTARGS=" -run TestAccOvirtVM_OperatingSystem"
```

### CODE REVIEW

**Comment by rgolangh** [comment]
@imjoey PTAL

File: ovirt/resource_ovirt_vm.go [hunk 1]
```diff
@@ -261,6 +276,15 @@ func resourceOvirtVM() *schema.Resource {
 					},
 				},
 			},
+			"type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Description: fmt.Sprintf(
```
Comment (imjoey) [inline ovirt/resource_ovirt_vm.go]: Hi @rgolangh , shall we define an extra `ValidateFunc` function to validate enum properties for fail-fast? Thanks.
```golang
    ValidateFunc: validation.StringInSlice([]string{
        string(ovirtsdk4.VMTYPE_DESKTOP),
        string(ovirtsdk4.VMTYPE_SERVER),
        string(ovirtsdk4.VMTYPE_HIGH_PERFORMANCE),
    }, false),
```

Comment (rgolangh) [reply ovirt/resource_ovirt_vm.go]: by all means.

Comment (rgolangh) [reply ovirt/resource_ovirt_vm.go]: ack

Comment (imjoey) [reply ovirt/resource_ovirt_vm.go]: @rgolangh thank you.

File: ovirt/resource_ovirt_vm_test.go [hunk 2]
```diff
@@ -596,3 +597,21 @@ resource "ovirt_vm" "vm" {
...
+
+var osCases = []OSTestParams{
+	{
+		"rhcos_x86",
+		true,
+	},
+}
+
+func TestVMOperatingSystemExpansion(t *testing.T) {
+	os, err := expandOS([]interface{}{osCases})
```
Comment (imjoey) [inline ovirt/resource_ovirt_vm_test.go]: This makes CI failed with message `ovirt/resource_ovirt_vm_test.go:614:35: cannot use []interface {} literal (type []interface {}) as type *schema.ResourceData in argument to expandOS`. Could you please fix this? Thanks.

Comment (rgolangh) [reply ovirt/resource_ovirt_vm_test.go]: done

**Comment by imjoey** [comment]
Look good to me. Thank you very much. @rgolangh .
Actually the CI job https://travis-ci.org/github/oVirt/terraform-provider-ovirt/jobs/666778310 has finished successfully. So let us merge it.

**Pull request merged**